### PR TITLE
PROXY_TYPE_SOCKS5 with str user/pass raised TypeError

### PR DESCRIPTION
**Pull request by temoto**
https://github.com/httplib2/httplib2/issues/144
https://github.com/httplib2/httplib2/pull/115

### CODE REVIEW

**Comment by temoto** [comment]
It was fixed in #115 for one proxy type (HTTP), now for all.

**Comment by temoto** [comment]
@stinky2nine Alex I reckon these new changes warrant another review.

File: tests/__init__.py [hunk 1]
```diff
@@ -295,7 +296,7 @@ def server_socket_thread(srv):
                 )
         except Exception as e:
             # traceback.print_exc caused IOError: concurrent operation on sys.stderr.close() under setup.py test
-            sys.stderr.write(traceback.format_exc().encode())
+            print(traceback.format_exc(), file=sys.stderr)
```
Comment (stinky2nine) [inline tests/__init__.py]: logging.warning or .error?

Comment (temoto) [reply tests/__init__.py]: It's test infrastructure. I'd rather avoid logging there. Do you insist?

Comment (stinky2nine) [reply tests/__init__.py]: Print statements are sort of the worst for everything. It's also for consistency.

What's the intention for printing the error? Perhaps just delete this line?

Comment (temoto) [reply tests/__init__.py]: Yes, intention is not clear. It seems I suck at context managers with regard to error handling. Problem summarized in this pseudo-snippet:

```
@contextmanager
def helper():
  shared = [None]  # cross-thread variable sharing
  t = Thread(catches_exception_into_shared)
  t.start()
  yield "to test code"
  # anything below is not executed
  t.join()
  if shared[0]:
    raise shared[0]  # so raise is also not executed
```

**Pull request merged**